controllers/admin/AdminImportController.php

```diff
@@ -3682,30 +3697,49 @@ public function storeContactImport($offset = false, $limit = false, $validateOnl
             );
         }
         $this->closeCsvFile($handle);
+
+        return $line_count;
     }
 
-    public function storeContactImportOne($info, $shop_is_feature_active, $regenerate, $force_ids, $validateOnly = false)
+    /**
+     * Import a store.
+     * If passed data contains a known store id, this store will be updated.
+     * Can also be used for store data validation only.
+     *
+     * @param array $info         : Store data
+     * @param       $notUsed      : Not used anymore.
+     * @param bool  $regenerate   : If images should be regenerated
+     * @param bool  $forceIds     : If passed store id should be used for store creation
+     * @param bool  $validateOnly : If set to true, store will not be updated nor created with $info data
+     *
+     * @return void
+     */
+    public function storeContactImportOne($info, $notUsed, $regenerate, $forceIds, $validateOnly = false)
     {
         AdminImportController::setDefaultValues($info);
 
-        if ($force_ids && isset($info['id']) && (int)$info['id']) {
-            $store = new Store((int)$info['id']);
-        } else {
-            if (array_key_exists('id', $info) && (int)$info['id'] && Store::existsInDatabase((int)$info['id'], 'store')) {
-                $store = new Store((int)$info['id']);
-            } else {
-                $store = new Store();
-            }
-        }
+        $storeId = isset($info['id']) ? (int)$info['id'] : null;
+        $store   = new Store($storeId);
+
+        $store->force_id = (bool)$forceIds;
 
         AdminImportController::arrayWalk($info, array('AdminImportController', 'fillInfo'), $store);
 
-        if (isset($store->image) && !empty($store->image)) {
-            if (!(AdminImportController::copyImg($store->id, null, $store->image, 'stores', !$regenerate))) {
-                $this->warnings[] = $store->image.' '.$this->trans('cannot be copied.', array(), 'Admin.Advparameters.Notification');
+        if (!empty($store->image)) {
+            $imgWasCopied = AdminImportController::copyImg($store->id, null, $store->image, 'stores', !$regenerate);
```

Comment:
It was done to enhance readability of the code (and to break the original long which was too long)